[README] Update Windows instructions (#2036)
awslabs/amazon-kinesis-video-streams-webrtc-sdk-c: README.md

### Configuring on Windows

#### Installing the dependencies / environment

> Install [MS Visual Studio Community / Enterprise](https://visualstudio.microsoft.com/vs/community/), [Strawberry perl](https://strawberryperl.com/), and [Chocolatey](https://chocolatey.org/install) if not installed already

Reviewer: Am I able to use `winget` instead of `chocolatey`?

https://learn.microsoft.com/en-us/windows/package-manager/winget/

Author: We recommend using choco

> Install [MS Visual Studio Community / Enterprise](https://visualstudio.microsoft.com/vs/community/), [Strawberry perl](https://strawberryperl.com/), and [Chocolatey](https://chocolatey.org/install) if not installed already

Reviewer 2: ```suggestion
Install [MS Visual Studio Community / Enterprise](https://visualstudio.microsoft.com/vs/community/), [Strawberry perl](https://strawberryperl.com/), and [Chocolatey](https://chocolatey.org/install) if not installed already.
```

> Get the libraries by running the following in `powershell`

Reviewer 2: ```suggestion
Install dependency libraries by running the following in `powershell`
```

Reviewer 2: Left as is, this line sounds likes it's talking about the above MSVC, Strawberry, Chocolatey "libraries", but it is not.

Author: It has the instructions right below. I don't think this can be misunderstood

Reviewer 2: Requesting the change mostly for grammar/correctness sake. Online search yields that "the" should be used if "The noun is known to the speaker" or "The noun has been mentioned before".

Another approach could be to change it to:
"Get the following libraries by..."
This one is OK by me from a correctness standpoint but lacks the context that these are dependency libraries.

Author: Here, "the libraries" refers to the ones mentioned in the header

```shell
choco install pkgconfiglite
choco install gstreamer
choco install gstreamer-devel
curl.exe -o C:\tools\pthreads-w32-2-9-1-release.zip ftp://sourceware.org/pub/pthreads-win32/pthreads-w32-2-9-1-release.zip
mkdir C:\tools\pthreads-w32-2-9-1-release\
Expand-Archive -Path C:\tools\pthreads-w32-2-9-1-release.zip -DestinationPath C:\tools\pthreads-w32-2-9-1-release
```
#### Customizing the `.github\build_windows_openssl.bat` file
If you unzipped at a path other than the one mentioned above / your pkg-config is in a different location, modify the path to the downloaded and unzipped PThreads in cmake in `.github\build_windows_openssl.bat`
```shell

> cmake -G "NMake Makefiles" -DBUILD_TEST=TRUE -DENABLE_AWS_SDK_IN_TESTS=OFF -DPKG_CONFIG_EXECUTABLE="D:\\gstreamer\\1.0\\x86_64\\bin\\pkg-config.exe" -DEXT_PTHREAD_INCLUDE_DIR="C:/tools/pthreads-w32-2-9-1-release/Pre-built.2/include/" -DEXT_PTHREAD_LIBRARIES="C:/tools/pthreads-w32-2-9-1-release/Pre-built.2/lib/x64/libpthreadGC2.a" ..

Reviewer: Why should we be building the tests?

Author: Removed

```

If not using MSVC Enterprise 2022 or have MSVC in a different location modify the path to MSVC as well in the `.github\build_windows_openssl.bat`

##### Enterprise
```shell
call "C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Auxiliary\Build\vcvarsall.bat" x86_amd64
```

##### Community
```shell
call "C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Auxiliary\Build\vcvarsall.bat" x86_amd64
```

##### Custom version
```shell
call "C:\Program Files\Microsoft Visual Studio\<your-version>\Community\VC\Auxiliary\Build\vcvarsall.bat" x86_amd64
```

#### Allowing long paths
Allow long paths before we start the build
```shell
git config --system core.longpaths true
```

Note that if the paths are still too long (which can cause the build to fail unfortunately), we recommend renaming the folders to use shorter names and moving them to `C:/`

#### Building the SDK

> Run the script customized above

Reviewer 2: ```suggestion
Run the `build_windows_openssl.bat` file script customized above
```

Reviewer 2: Using consistent naming across different sections is important. If a reader is scanning through and they see this line, they will quickly be able to connect it to where we introduced this script in the previous "Customizing the .github\build_windows_openssl.bat file" section. We do not at all refer to it as a script there, so adding the longer name of it here too.

Also can consider removing "customized above" as the customization part is optional and not likely done by most first-time users - or can replace it with "... script which can be customized using the above instructions."

Author: The previous step mentions which file we were trying to customize and the step to build already mentions `.github\build_windows_openssl.bat` so I think it is clear

```shell
.github\build_windows_openssl.bat
```

#### Customizing and setting PATH to run the samples
To run the sample application, make sure that you've exported the following paths and appended them to env:Path for `powershell`.

> Modify the `<path-to-webrtc-root>\open-source\bin`, `<path-to-webrtc-root>\build`, `<path-to-pthreads-unzip-location>\pthreads-w32-2-9-1-release\Pre-built.2\dll\x64`

Reviewer 2: This is unclear since there is no place where we have a `<path-to-webrtc-root>` to be modified. Please consider adding `<path-to-webrtc-root>` to the below command. Can keep a filled in one too as an example.

Author: Done

> Modify the `<path-to-webrtc-root>\open-source\bin`,`<path-to-webrtc-root>\build`, `<path-to-pthreads-unzip-location>\pthreads-w32-2-9-1-release\Pre-built.2\dll\x64`

Reviewer: Spacing

#### Customizing the `.github\build_windows_openssl.bat` file
If you unzipped at a path other than the one mentioned above / your pkg-config is in a different location, modify the path to the downloaded and unzipped PThreads in cmake in `.github\build_windows_openssl.bat`
```shell

> cmake -G "NMake Makefiles" -DENABLE_AWS_SDK_IN_TESTS=OFF -DPKG_CONFIG_EXECUTABLE="D:\\gstreamer\\1.0\\x86_64\\bin\\pkg-config.exe" -DEXT_PTHREAD_INCLUDE_DIR="C:/tools/pthreads-w32-2-9-1-release/Pre-built.2/include/" -DEXT_PTHREAD_LIBRARIES="C:/tools/pthreads-w32-2-9-1-release/Pre-built.2/lib/x64/libpthreadGC2.a" ..

Reviewer: Is this needed?

`-DENABLE_AWS_SDK_IN_TESTS=OFF` ?
